### PR TITLE
kexec support

### PR DESCRIPTION
**Pull request by russell-peterson**
Add basic helper scripts for doing a simple reboot
using kexec.

bfcores-online [<count>] -- leave <count> cores online
                            take the rest offline
bfkexec-load             -- load the desired kernel image
                            using kexec
bfkexec-reboot [<image>] -- reboot the system using kexec
                            (/boot/Image by default)

### CODE REVIEW

File: bfcores-online [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  echo "must leave at least 1 core online"
+  exit 1
+fi
+
+if [ $1 -ge $total_cores ]; then
+  echo "requested number of cores too large"
+  exit 1
+fi
+
+offline_cores=$(($total_cores-$1-1))
```
Comment (lsun100) [inline bfcores-online:44]: Q: why need the '-1' here?

File: bfkexec-load [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+KEXEC_LOADED=$(cat /sys/kernel/kexec_loaded)
+
+if [ "$KEXEC_LOADED" -eq 1 ]; then
+  echo kernel already loaded
+  exit 0
+fi
+
+if [ -n "$1" ]; then
+  KERNEL_IMAGE="/boot/Image"
+elif
```
Comment (lsun100) [inline bfkexec-load:39]: else (not elif)

File: bfkexec-load [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  KERNEL_IMG="$1"
```
Comment (lsun100) [inline bfkexec-load:40]: When it comes to this line, '$1' is empty. Maybe line 37 means [ -z "$1" ]?

**Comment by russell-peterson** [comment]
On further review these new helper scripts are not required.

**Pull request closed without merging**